src/main/java/org/cryptomator/ui/preferences/UpdatesPreferencesController.java

```diff
@@ -1,48 +1,104 @@
 	private final ObjectBinding<ContentDisplay> checkForUpdatesButtonState;
 	private final ReadOnlyStringProperty latestVersion;
 	private final String currentVersion;
 	private final BooleanBinding updateAvailable;
+	private final ObjectProperty<LocalDateTime> updateCheckDateProperty;
+	private final Comparator<String> versionComparator = new SemVerComparator();
+	private final ObjectProperty<UpdateChecker.UpdateCheckState> updateCheckStateProperty;
 
 	/* FXML */
 	public CheckBox checkForUpdatesCheckbox;
+	public FormattedLabel updateCheckDateFormattedLabel;
+	public HBox checkFailedHBox;
+	public FormattedLabel latestVersionFormattedLabel;
+	public Label upToDateLabel;
 
 	@Inject
-	UpdatesPreferencesController(Application application, Settings settings, UpdateChecker updateChecker) {
+	UpdatesPreferencesController(Application application, Environment environment, Settings settings, UpdateChecker updateChecker) {
 		this.application = application;
+		this.environment = environment;
 		this.settings = settings;
 		this.updateChecker = updateChecker;
 		this.checkForUpdatesButtonState = Bindings.when(updateChecker.checkingForUpdatesProperty()).then(ContentDisplay.LEFT).otherwise(ContentDisplay.TEXT_ONLY);
 		this.latestVersion = updateChecker.latestVersionProperty();
-		this.updateAvailable = latestVersion.isNotNull();
 		this.currentVersion = updateChecker.getCurrentVersion();
+		this.updateAvailable = Bindings.createBooleanBinding(() -> {
+			if (latestVersion.get() != null) {
+				return versionComparator.compare(currentVersion, latestVersion.get()) < 0;
+			} else {
+				return false;
+			}
+		}, latestVersion);
+		this.updateCheckDateProperty = updateChecker.updateCheckTimeProperty();
+		this.updateCheckStateProperty = updateChecker.updateCheckStateProperty();
 	}
 
 	public void initialize() {
 		checkForUpdatesCheckbox.selectedProperty().bindBidirectional(settings.checkForUpdates);
+
+		DateTimeFormatter formatter = DateTimeFormatter.ofLocalizedDateTime(FormatStyle.MEDIUM).withLocale(Locale.getDefault());
+		updateCheckDateFormattedLabel.arg1Property().bind(Bindings.createStringBinding(() -> (!updateCheckDateProperty.get().equals(LocalDateTime.parse(Settings.DEFAULT_LAST_UPDATE_CHECK)) && latestVersionProperty().isNotNull().get()) ? updateCheckDateProperty.get().format(formatter) : "-", updateCheckDateProperty, latestVersionProperty()));
+
+		BooleanBinding isUpdateCheckFailed = updateCheckStateProperty.isEqualTo(UpdateChecker.UpdateCheckState.CHECK_FAILED);
+		checkFailedHBox.managedProperty().bind(isUpdateCheckFailed);
+		checkFailedHBox.visibleProperty().bind(isUpdateCheckFailed);
+
+		latestVersionFormattedLabel.arg1Property().bind(Bindings.createStringBinding(() -> (latestVersion.get() != null) ? latestVersion.get() : "-", latestVersion));
+
+		BooleanBinding isUpdateSuccessfulAndCurrent = updateCheckStateProperty.isEqualTo(UpdateChecker.UpdateCheckState.CHECK_SUCCESSFUL).and(latestVersion.isEqualTo(currentVersion));
```

Comment:
move as many binding operations to the fxml and create in the controller just bindings which are returned as getter/setter